src/Jellyfin.Maui/Services/StateStorageService.cs

```diff
@@ -121,41 +112,38 @@ public async ValueTask AddUserAsync(UserStateModel userStateModel)
     public async ValueTask RemoveServerAsync(Guid serverId)
     {
         var state = await GetStoredStateAsync().ConfigureAwait(false);
-        for (var index = 0; index < state.Servers.Count; index++)
+
+        var server = state.Servers.FirstOrDefault(x => x.Id == serverId);
+
+        if (server is null)
         {
-            if (state.Servers[index].Id == serverId)
-            {
-                // Remove all linked users.
-                for (var userIndex = state.Users.Count; userIndex >= 0; userIndex--)
-                {
-                    if (state.Users[userIndex].ServerId == serverId)
-                    {
-                        state.Users.RemoveAt(userIndex);
-                    }
-                }
-
-                state.Servers.RemoveAt(index);
-                break;
-            }
+            return;
+        }
+
+        foreach (var user in state.Users.Where(x => x.ServerId == serverId).ToList())
```

Comment:
```suggestion
        foreach (var user in state.Users.Where(x => x.ServerId == serverId))
```

`.ToList()` shouldn't be required here